### PR TITLE
fix(database) failed to initialize database core

### LINKED ISSUE #308: failed to initialize database core
[ORM]2020/12/09 08:48:20 register db Ping `default`, pq: database "core" does not exist
2020-12-09T08:48:20Z [FATAL] [/core/main.go:120]: failed to initialize database: register db Ping `default`, pq: database "core" does not exist

image

### PR DESCRIPTION
**Pull request by Ghostbaby**
failed to initialize database core

fix #308

Signed-off-by: zhuhuijun <zhuhuijunzhj@gmail.com>

### CODE REVIEW

File: pkg/cluster/controllers/database/readiness.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ const (
 	ClairDatabase        = "clair"
 	NotaryServerDatabase = "notaryserver"
 	NotarySignerDatabase = "notarysigner"
+	DefaultDatabaseName  = "postgres"
```
Comment (steven-zou) [inline pkg/cluster/controllers/database/readiness.go:31]:
```suggestion
	DefaultDatabaseUser  = "postgres"
```

File: pkg/cluster/controllers/database/utils.go [hunk 2]
```diff
@@ -17,12 +16,12 @@ const (
...
 	databases := map[string]string{
-		CoreDatabase: harbormetav1.CoreDatabase,
+		CoreDatabase: DefaultDatabaseName,
 	}
 
 	if p.HarborCluster.Spec.Notary != nil {
-		databases[NotaryServerDatabase] = harbormetav1.NotaryServerDatabase
-		databases[NotarySignerDatabase] = harbormetav1.NotarySignerDatabase
+		databases[NotaryServerDatabase] = DefaultDatabaseName
+		databases[NotarySignerDatabase] = DefaultDatabaseName
```
Comment (steven-zou) [inline pkg/cluster/controllers/database/utils.go:24]: Check the code:

```go
func GetUsers() map[string]api.UserFlags {
	return map[string]api.UserFlags{
		"zalando": {
			"superuser",
			"createdb",
		},
		"foo_user": {},
	}
}
```

you did not create `DefaultDatabaseName` here, the user you created is `zalando`.

**Review by steven-zou — COMMENTED** [review]
commented

**Review by steven-zou — APPROVED** [review]
LGTM

**Review by chlins — APPROVED** [review]
LGTM

**Pull request merged**